### DISCUSSION TITLE
One-off First run page, how to implement it?

### DISCUSSION THREAD

**Discussion by jkommeren** [dotnet/maui#18402] [Q&A]
So the idea is simple (I'd say) but I cannot find a clean way to implement it in MAUI. I also cannot find any examples :(!

I want to have a getting started page that's only visible on first run, or when configuration has been cleared. After that page has been completed (i.e. the user has signed in) it should no longer be accessible.

So two cases:
First run:
- Access to Getting Started page
- No access to other pages in app (also not visible in Shell flyout), until successfully signed in.

And all starts after it:
- Access to all other pages
- Getting Started not visible/accessible.

I've been using a Popup until now, from the community toolkit, but the popup seems to disappear in many cases (move to different monitor in Windows, or switching to another app in Android), revealing the Shell beneath.
Update: to make things worse there is no way to detect the popup disappearing (i.e. closed event is not fired) so I'm afraid I cannot rely on that any longer.

I could imagine making pages visible / invisible in the Shell somehow, but that feels a bit clumsy.

(Or is this idea I have bad practice in general? How do you guys do this?)

**Comment by Xyncgas** [comment]
Create your page, use a variable as toggle in the binding I guess
Or use MAUI.Blazor and there's a million javascript library to do it

**Reply by jkommeren** [reply]
> Create your page, use a variable as toggle in the binding I guess
> Or use MAUI.Blazor and there's a million javascript library to do it

Thanks for your response. So your suggestion is to include it in the appshell xaml right?

Just want to make sure I'm not missing something obvious 

Maui blazor is a valid option I guess. I just don't like JavaScript 😂

**Reply by jkommeren** [reply]
Ah I see my issue. I used Shell.gotoasync, which required the page to be defined in appshell, instead of navigation.pushasync (or pushmodal, which seems to be the right candidate for me!)

Mystery solved 😅
